.travis.yml

```
@@ -1,83 +1,95 @@
-# NOTE: Please try to select up to six instances that add some meaningful
-#       testing of the formula's behaviour. If possible, try to refrain from
-#       the classical "chosing all the instances because I want to test on
-#       another/all distro/s" trap: it will just add time to the testing (see
-#       the discussion on #121).  As an example, the set chosen below covers
-#       the most used distros families, systemd and non-systemd and the latest
-#       three supported Saltstack versions with python2 and 3.
-#       As for `kitchen.yml`, that should still contain all of the platforms,
-#       to allow for comprehensive local testing
-#       Ref: https://github.com/saltstack-formulas/template-formula/issues/118
-#       Ref: https://github.com/saltstack-formulas/template-formula/issues/121
-env:
-  matrix:
-    - INSTANCE: default-debian-10-develop-py3
-    # - INSTANCE: default-ubuntu-1804-develop-py3
-    # - INSTANCE: default-centos-7-develop-py3
-    # - INSTANCE: default-fedora-30-develop-py3
-    # - INSTANCE: default-opensuse-leap-15-develop-py3
-    # - INSTANCE: default-amazonlinux-2-develop-py2
-    # - INSTANCE: default-arch-base-latest-develop-py2
-    # - INSTANCE: default-debian-9-2019-2-py3
-    - INSTANCE: default-ubuntu-1804-2019-2-py3
-    # - INSTANCE: default-centos-7-2019-2-py3
-    # - INSTANCE: default-fedora-30-2019-2-py3
-    # - INSTANCE: default-opensuse-leap-15-2019-2-py3
-    - INSTANCE: default-amazonlinux-2-2019-2-py2
-    - INSTANCE: default-arch-base-latest-2019-2-py2
-    # - INSTANCE: default-debian-9-2018-3-py2
-    # - INSTANCE: default-ubuntu-1604-2018-3-py2
-    # - INSTANCE: default-centos-7-2018-3-py2
-    - INSTANCE: default-fedora-29-2018-3-py2
-    - INSTANCE: default-opensuse-leap-15-2018-3-py2
-    # - INSTANCE: default-amazonlinux-2-2018-3-py2
-    # - INSTANCE: default-arch-base-latest-2018-3-py2
-    # - INSTANCE: default-debian-8-2017-7-py2
-    # - INSTANCE: default-ubuntu-1604-2017-7-py2
-    - INSTANCE: centos6-centos-6-2017-7-py2
-    # - INSTANCE: default-fedora-29-2017-7-py2
-    # - INSTANCE: default-opensuse-leap-15-2017-7-py2
-    # - INSTANCE: default-amazonlinux-2-2017-7-py2
-    # - INSTANCE: default-arch-base-latest-2017-7-py2
+# Language and cache config
+language: ruby
+cache: bundler
 
+# Script to run for the test stage
 script:
-  - bin/kitchen verify ${INSTANCE}
+  - bin/kitchen verify "${INSTANCE}"
 
+# Stages and jobs matrix
+stages:
+  - test
+  - name: release
+    if: branch = master AND type != pull_request
 jobs:
   include:
-    # Define the `lint` stage (runs `yamllint` and `commitlint`)
-    - stage: lint
-      language: node_js
+    # Define the test stage matrix
+    # Run all of the linters in a single instance (concurrently with the main tests)
+    # Define the rest of the matrix based on Kitchen testing
+
+    - language: node_js
       node_js: lts/*
+      env: 'Lint: salt-lint, yamllint, rubocop & commitlint'
```

```suggestion
      name: 'Lint: salt-lint, yamllint, rubocop & commitlint'
```
What about this (for aesthetic reasons :smiley:)?

---

@dafyddj I did a lot of digging through Travis docs.  I can't recall the exact place but `env` is used for certain types of control, such as using the `allow_failures` feature.  I've tested that and I'm sure it will come in useful along the way, perhaps where we have certain lint checks that can be allowed to fail.  I haven't checked for `name` specifically, I'm not sure if it will work in the same way.  We can always revisit this later.

---

Fair enough; `name` just affects how the job appears in the UI, which is what I thought you were going for.

---

We _could_ use `name` alongside the `env`:

* https://travis-ci.org/myii/template-formula/jobs/595218889

Compared to:

* https://travis-ci.com/saltstack-formulas/template-formula/jobs/243474795

Not really feeling it for the main page, though:

* https://travis-ci.org/myii/template-formula/builds/595218888

I'll unresolve this for the time being, in case there are any comments about this.  Thanks again for the suggestions.

---

No problem. Either way it's hardly the issue of the century :wink:.

---

@dafyddj Nope, on reflection, you're right again.  I've updated the PR _and_ I've added a little more info to the `release` stage, so that it's more informative too.  General appearance (slight modifications since then):

* https://travis-ci.org/myii/template-formula/builds/595255179

Great reviewing, by the way.  Appreciate all of the feedback to tidy this up before it gets splurged everywhere!